metrics/lib/common.bash

```diff
@@ -42,6 +42,27 @@ function save_results(){
 	bash $LIB_DIR/send_results.sh -n "$1" -a "$2" -r "$3" -u "$4"
 }
 
+# This function checks existence of commands.
+# They can be received standalone standalone or as an array, e.g.
+#
+# cmds=(“cmd1” “cmd2”)
+# check_cmds "${cmd[@]}"
+function check_cmds()
+{
+	set +e
```

Comment:
Having said all that, and now looked at the code, we don't need the "set [+-]e" at all if we just rewrite the check to be part of the "if" statement.

We can loose the "function" keyword, this is not POSIX syntax.
Lets see if I can do markdown correctly.

    check_cmds()
    {
         for cmd ; do
              if ! command -v "$cmd" >/dev/null 2>&1 ; then
                   die "command $cmd not available"
                   exit 1
              fi
              printf 'command: %s: yes\n' "$cmd"
          done
     }